File: Sample/Sample/InboxDelegate.m
/* Copyright Urban Airship and Contributors */

#import "InboxDelegate.h"
#import "MessageCenterViewController.h"
#import "AppDelegate.h"

@interface InboxDelegate ()
@property(nonatomic, strong) UITabBarController *tabBarController;
@property(nonatomic, strong) MessageCenterViewController *messageCenterViewController;
@end

@implementation InboxDelegate

- (instancetype)initWithRootViewController:(UIViewController *)rootViewController {
    self = [super init];
    if (self) {
        self.tabBarController = (UITabBarController *)rootViewController;

        self.messageCenterViewController = [self.tabBarController.viewControllers objectAtIndex:MessageCenterTab];
    }
    return self;
}

- (void)showInbox {
    dispatch_async(dispatch_get_main_queue(), ^{
        self.tabBarController.selectedIndex = MessageCenterTab;
        [self.messageCenterViewController showInbox];
    });
}

- (void)showMessageForID:(NSString *)messageID {
    [self showInbox];
    dispatch_async(dispatch_get_main_queue(), ^{
        [self.messageCenterViewController displayMessageForID:messageID];
    });
}

@end

File: SwiftSample/SwiftSample/InboxDelegate.swift
/* Copyright Airship and Contributors */

import Foundation
import AirshipKit

class InboxDelegate : NSObject, UAInboxDelegate {

    var tabBarController : UITabBarController;
    var messageCenterViewController : MessageCenterViewController;

    init(tabBarController:UITabBarController) {
        self.tabBarController = tabBarController
        let appDelegate = UIApplication.shared.delegate as! AppDelegate
        self.messageCenterViewController = self.tabBarController.viewControllers![appDelegate.MessageCenterTab] as! MessageCenterViewController;
    }

    func showInbox() {
        DispatchQueue.main.async {
            let appDelegate = UIApplication.shared.delegate as! AppDelegate
            self.tabBarController.selectedIndex = appDelegate.MessageCenterTab
            self.messageCenterViewController.showInbox()
        }
    }

    func showMessage(forID messageID: String) {
        self.showInbox()
        DispatchQueue.main.async {
            self.messageCenterViewController.displayMessageForID(messageID)
        }
    }
}


